### ISSUE TITLE
Inclusive gateway only allows default flow with condition

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#10310]
**Describe the bug**

According to the spec (10.5.3 - page 293):
>The default Sequence Flow should not have a conditionExpression.
>Any such Expression SHALL be ignored.

However, processes with a default flow are required to have a condition in Zeebe currently, and this condition is evaluated.

**To Reproduce**

- deploy a process with inclusive gateway with multiply outgoing sequence flows with conditions and a default outgoing sequence flow
- notice the rejection

**Expected behavior**

- Inclusive gateway's default flow should be allowed to have a condition, but not be required
- Zeebe should not evaluate this condition.

**Comment by korthout** [comment]
We should fix this before the 8.1 release.

**Comment by skayliu** [comment]
Hi,@korthout ,you can  assigne this issue to me.

**Comment by korthout** [comment]
@skayliu Of course! Thanks 🎖️

**Comment by skayliu** [comment]
> **Expected behavior**
> 
> * Inclusive gateway's default flow should be allowed to have a condition, but not be required
> * Zeebe should not evaluate this condition.

Hi,@korthout, according to this, do we need to change the doc?

**Comment by korthout** [comment]
> do we need to change the doc?

Yes we should also change the doc. Specifically we should make the following changes:
- in the screenshots, the default flow's are labeled (somewhat indicating that there is a condition). We should remove the labels on the default flow
- it states that `If an inclusive gateway has multiple outgoing sequence flows, all sequence flows must have a condition to define when the flow is taken. Optionally, one of the sequence flows can be marked as the default flow. ` which is no longer true. We should indicate that the default flow does not require a condition and that it would be ignored if it does have one
- the screenshot `inclusive-gateway-1` indicates that the condition of the default flow was evaluated. We need to change that
- the same is true for screenshot `inclusive-gateway-2`

**Comment by korthout** [comment]
@skayliu No problem. I'll have a look at it later today 🙂 Thanks for all your efforts on this topic ❤️

**Comment by philippfromme** [comment]
@korthout Looking at the validation of `bpmn:ExclusiveGateway` elements (cf. https://github.com/camunda/zeebe/blob/clients/go/v8.1.3/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/ExclusiveGatewayValidator.java#L38) a default flow whose source is an exclusive gateway is still not allowed to have a condition. Shouldn't this also be changed in accordance with the BPMN 2.0 specification?

I'm asking because I'm working on a lint rule that checks outgoing sequence flows of exclusive gateways (cf. https://github.com/camunda/bpmnlint-plugin-camunda-compat/issues/57).

**Comment by korthout** [comment]
@philippfromme Seems you're right about that. Note that this issue was rectifying incorrect behavior for the inclusive gateway, where it really matters whether or not the condition on the default flow is evaluated. The previous implementation of OR required the default flow to have a condition.

The behavior for the exclusive gateway is less severe. We first check all the other conditions, then take the default flow if none was met. We already do this now as well, but simply don't allow setting a condition on the default flow. 

I guess rejecting this for XOR is simply historical (easier to implement this way), but we should align them at some point. Feel free to open an issue for it.

**Comment by korthout** [comment]
>when I create a diagram with a default flow whose source is an exclusive gateway and then add a condition to that sequence flow [..] my deployment would fail

That's correct. Zeebe will reject this at this time. We may loosen this restriction in the future to align it with the inclusive gateway (and the spec), as you suggested.

**Comment by skayliu** [comment]
Hi,@korthout, if you don't mind, I can make a PR for this.

**Comment by philippfromme** [comment]
As the user cannot easily create such a diagram, it's not urgent. 👍🏻

**Comment by korthout** [comment]
@skayliu Please open an issue for it first, but go ahead. I would accept a PR for it 👍 

Please note that in addition to removing the validation, we also have to make some adjustments to the existing code to avoid evaluating the Condition on the Default flow.
